src/temp/ClauseExample.java

```diff
@@ -0,0 +1,33 @@
+package temp;
+
+import global.BaseTest;
+import global.variables.*;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ClauseExample extends BaseTest {
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("This is a random test with retrieval."),
+                new NewLine(),
+                new Optional(new RandomInteger(0, 10)),
+        };
+    }
+
+    public void runMain() {
+        Example.main(new String[]{});
+    }
+
+    @RepeatedTest(50)
+    public void retrieveValueTest() {
+//        int randomIntegerValue = Integer.parseInt(getItemAtIndex(3));
+        if (getItemAtIndex(3) != null) {
+            int randomIntegerValue = Integer.parseInt(getItemAtIndex(3));
+            System.out.println(randomIntegerValue);
+            assertEquals(11, randomIntegerValue);
```

Comment:
Wait what does this test do? Does it assert that the randomIntegerValue should be 11?  but won't it be a random integer from 0 to 9? So this test would fail always?

Reply:
Yeah this is just test code so I wanted it to fail so I could see the value it retrieved

Reply:
Okay perfect